File: backend/src/main/java/com/back/domain/member/member/dto/MemberWithUsernameDto.java
package com.back.domain.member.member.dto;

import com.back.domain.member.member.entity.Member;
import org.springframework.lang.NonNull;

import java.time.LocalDateTime;

public record MemberWithUsernameDto(
        @NonNull int id,
        @NonNull LocalDateTime createDate,
        @NonNull LocalDateTime modifyDate,
        @NonNull String name,
        @NonNull String username,
        @NonNull boolean isAdmin,
        @NonNull String profileImageUrl
) {
    public MemberWithUsernameDto(int id, LocalDateTime createDate, LocalDateTime modifyDate, String name, String username, boolean isAdmin, String profileImageUrl) {
        this.id = id;
        this.createDate = createDate;
        this.modifyDate = modifyDate;
        this.name = name;
        this.username = username;
        this.isAdmin = isAdmin;
        this.profileImageUrl = profileImageUrl;
    }

    public MemberWithUsernameDto(Member member) {
        this(
                member.getId(),
                member.getCreateDate(),
                member.getModifyDate(),
                member.getName(),
                member.getUsername(),
                member.isAdmin(),
                member.getProfileImgUrlOrDefault()
        );
    }
}

File: backend/src/main/java/com/back/domain/member/member/dto/MemberWithUsernameDto.kt
package com.back.domain.member.member.dto

import com.back.domain.member.member.entity.Member
import java.time.LocalDateTime


data class MemberWithUsernameDto(
    val id: Int,
    val createDate: LocalDateTime,
    val modifyDate: LocalDateTime,
    val username : String,
    val name: String,
    val isAdmin: Boolean,
    val profileImageUrl: String
) {
    constructor(member : Member) : this(
        member.id,
        member.createDate,
        member.modifyDate,
        member.username,
        member.name,
        member.isAdmin,
        member.profileImgUrlOrDefault
    )
}
